Modules/_datetimemodule.c

```diff
@@ -680,6 +713,11 @@ set_date_fields(PyDateTime_Date *self, int y, int m, int d)
  * String parsing utilities and helper functions
  */
 
+static unsigned char
+is_digit(const char c) {
+    return ((unsigned int)(c - '0')) < 10;
```

Comment:
What about Unicode digits? Or does the spec you're implementing only allow ASCII?

Reply:
The spec only allows ASCII. We go beyond the spec in a few places, but the only place where a unicode character can be introduced is the separator, and the separator should never really be a "digit" even if it's a unicode digit.